nxc/protocols/mssql.py

```diff
@@ -297,42 +297,53 @@ def mssql_query(self):
 
     @requires_admin
     def execute(self, payload=None, get_output=False):
-        if not payload and self.args.execute:
-            payload = self.args.execute
-
-        if not self.args.no_output:
-            get_output = True
-
-        self.logger.info(f"Command to execute: {payload}")
+        payload = self.args.execute if not payload and self.args.execute else payload
+        if not payload:
+            self.logger.error("No command to execute specified!")
+            return None
+        
+        get_output = True if not self.args.no_output else get_output
+        self.logger.debug(f"{get_output=}")
+        
         try:
             exec_method = MSSQLEXEC(self.conn, self.logger)
-            raw_output = exec_method.execute(payload, get_output)
+            output = exec_method.execute(payload)
+            self.logger.debug(f"Output: {output}")
         except Exception as e:
             self.logger.fail(f"Execute command failed, error: {e!s}")
             return False
         else:
-            self.logger.success("Executed command via mssqlexec")
-            if raw_output:
-                for line in raw_output:
-                    self.logger.highlight(line)
```

Comment:
Have to test this one, but without having "for line in x" it would display the whole message in one highlight blob or not?

Reply:
In contrast to smb this is indeed displayed as one highlight blob.
MSSQL:
![image](https://github.com/Pennyw0rth/NetExec/assets/61382599/4738b472-be51-4ef1-871c-b6bfa30a5fa8)
SMB:
![image](https://github.com/Pennyw0rth/NetExec/assets/61382599/4d406505-984a-46bb-8b04-89284dfeb987)